### PR TITLE
OIDC support

### PR DESCRIPTION
**Pull request by thomas11**
- Add OIDC configuration
- Add OIDC testing in CI

[CI test run with debug logging on to prove we're setting `OIDC_ARM_CLIENT_ID`](https://github.com/pulumi/pulumi-azure/actions/runs/4567390150/jobs/8064673591).

Supersedes #1165

### CODE REVIEW

File: .github/workflows/run-acceptance-tests.yml [hunk 1]
```diff
@@ -225,7 +225,7 @@ jobs:
     - name: Is workflow a success
       run: echo yes
   test:
-    if: github.event_name == 'repository_dispatch' ||
+    if: github.event_name == 'repository_dispatch' || github.event_name == 'workflow_dispatch' ||
```
[the remark this answers is not in the document]
Comment (thomas11) [reply .github/workflows/run-acceptance-tests.yml:228]: It adds the "Run workflow" button to the UI for manual runs.

[the remark this answers is not in the document]
Comment (thomas11) [reply .github/workflows/run-acceptance-tests.yml:228]: This part is optional for back porting to ci-mgmt.

[the remark this answers is not in the document]
Comment (t0yv0) [reply .github/workflows/run-acceptance-tests.yml:228]: Ah nice, so we just need to back port an env var. Throw a ticket into ci-mgmt and you can merge this without waiting on the backport I think.

[the remark this answers is not in the document]
Comment (thomas11) [reply .github/workflows/run-acceptance-tests.yml:228]: pulumi/ci-mgmt#382

**Review by kpitzen — APPROVED** [review]
LGTM! Wish the test output were a _little_ more readable post-hoc in GH, but it seems like it's working as expected.

**Pull request merged**